frontend/packages/console-plugin-sdk/src/typings/providers.ts

```diff
@@ -0,0 +1,20 @@
+import { ProviderExoticComponent } from 'react';
+import { Extension } from './base';
+
+namespace ExtensionProperties {
+  export interface ContextProvider<T> {
+    /** Context Provider Exotic Component. */
+    Provider: ProviderExoticComponent<T>;
+    /** Hook for the Context value. */
+    useValueHook: () => T;
```

Comment:
Since the hook isn't lazy loaded, this may play havoc with dependency resolution until we get extensibility based on JSON descriptors.

If so, we may need to make this promise based so that the code ref gets resolved lazily.

Reply:
Hmm..Hooks can't be loaded with `React.lazy`, this would mean create a new `Async` component for providers? or do you see an easier way?

Reply:
All code references in dynamic plugins will eventually become promise based and lazy loaded.
cc @vojtechszocs 

For now if this is an issue you can do `useValueHook: Promise<() => T>` however because context must be supplied up front, I don't see a need to enforce a promise. We will end up resolving all the promises as soon as the app loads.